lib/pbench/server/cache_manager.py

```diff
@@ -590,96 +590,42 @@ def create(
 
         return cls(destination, resource_id, controller)
 
-    def cache_map(self, dir_path: Path):
+    def build_map(self):
         """Build hierarchical representation of results tree
 
-        NOTE: this structure isn't removed when we release the cache, as the
-        data remains valid.
+        This must be called with the cache locked (shared lock is enough)
+        and unpacked.
 
-        Args:
-            dir_path: root directory
+        NOTE: this structure isn't removed when we release the cache, as the
+        data remains valid so long as the dataset exists.
         """
-        root_dir_path = dir_path.parent
-        cmap: CacheMap = {
-            dir_path.name: {"details": make_cache_object(root_dir_path, dir_path)}
+        cmap: CacheMapEntry = {
+            "details": CacheObject.create(self.unpacked, self.unpacked)
         }
-        dir_queue = deque(((dir_path, cmap),))
+        dir_queue = deque(((self.unpacked, cmap),))
         while dir_queue:
-            dir_path, parent_map = dir_queue.popleft()
-            tar_n = dir_path.name
-
+            unpacked, parent_map = dir_queue.popleft()
             curr: CacheMapEntry = {}
-            for l_path in dir_path.glob("*"):
-                tar_info = make_cache_object(root_dir_path, l_path)
-                curr[l_path.name] = {"details": tar_info}
-                if l_path.is_symlink():
-                    continue
-                if l_path.is_dir():
-                    dir_queue.append((l_path, curr))
-            parent_map[tar_n]["children"] = curr
+            for path in unpacked.glob("*"):
+                details = CacheObject.create(self.unpacked, path)
+                curr[path.name] = {"details": details}
```

Comment:
You're probably right. I found the naming and use of those two highly confusing and somewhat poorly defined. I considered getting rid of the `dict` looseness here and building real `CacheMap` and `CacheMapEntry` dataclasses, but didn't quite take it that far. I think that would be easier to follow since at least the IDE cares (a little).